src/node/http.ts

```diff
@@ -927,7 +927,29 @@ export class HttpServer {
     }
 
     // Must be authenticated to use the proxy.
-    route.provider.ensureAuthenticated(request)
+    if (!route.provider.authenticated(request)) {
+      // Attempt to determine if it's the user browsing the root and if so fall
+      // through to allow the login flow.
+      if (request.headers["content-type"] !== "application/json") {
+        switch (route.providerBase) {
```

Comment:
I'm confused by this, don't we pretty always want to fall through to the login flow if `text/html` is asked for but the user isn't authenticated?

Reply:
Yup, if there was a way to reliably detect the request was asking for HTML (and ideally also that it wasn't an xhr request) that would 100% be the way to do it. All this jank is to work around not being able to do that (at least as far as I know, I hope I'm wrong though).